visualize-admin/visualization-tool
# feat: Embed mode improvements

Closes #1747

This PR:
- adds a responsive iFrame mode (with a small JS script that listens for a custom resize event and adjusts iFrame height accordingly),
- adds a way to remove chart border via URL query param (`?disableBorder=true`). This is not yet possible to set via UI, but this will be done in a next iteration (the feature is needed now for a BLW dashboard).

## How to reproduce
1. Go to [this link](https://seleniumbase.io/w3schools/iframes).
2. Replace the HTML content with below snippet (copied from [this link](https://test.visualize.admin.ch/en/v/lQjE19b_FBql?dataSource=Prod) [and modified a bit to show the iframe border, so it's visible that it's not responsive]).
```html
<!DOCTYPE html>
<html>
<body>

<iframe src="https://test.visualize.admin.ch/en/embed/lQjE19b_FBql" style="border:1px solid #000; max-width: 100%" name="visualize.admin.ch" scrolling="no" frameborder="1" marginheight="0px" marginwidth="0px" height="640px" width="600px" allowfullscreen></iframe>

</body>
</html>
```
3. ❌ Resize the panel and see that the iframe does not resize dynamically.
4. ❌ See that the chart has a gray border.

## How to test
1. Go to [this link](https://seleniumbase.io/w3schools/iframes).
2. Replace the HTML content with below snippet (copied from [this link](https://visualization-tool-dhein6se5-ixt1.vercel.app/en/v/-Gs1rzUcZFz7?dataSource=Prod) [responsive embed mode, modified a bit to show the iframe border, so it's visible that it's responsive]).
```html
<!DOCTYPE html>
<html>
<body>

<iframe src="https://visualization-tool-dhein6se5-ixt1.vercel.app/en/embed/-Gs1rzUcZFz7?disableBorder=true" width="100%" height="0px" style="border:1px solid #000;" name="visualize.admin.ch"></iframe>
<script type="text/javascript">!function(){window.addEventListener("message", function (e) { if (e.data.type === "VISUALIZE-CHART-RESIZE") { document.querySelectorAll("iframe").forEach((iframe) => { if (iframe.contentWindow === e.source) { iframe.style.height = e.data.height + "px"; } }); } })}();</script>

</body>
</html>
```
3. ✅ Resize the panel and see that the iframe resizes dynamically.
4. Modify the style property to change the border to the original one: `border:1px #ffffff none;`.
5. ✅ See that the chart misses a border.
